virtme/commands/run.py

```diff
@@ -968,6 +969,43 @@ def cleanup_console_script():
     qemuargs.extend(["-device", "vhost-vsock-pci,guest-cid=%d" % args.port])
 
 
+def ssh_client(args):
+    if args.remote_cmd is not None:
+        exec_escaped = args.remote_cmd.replace('"', '\\"')
+        remote_cmd = ["bash", "-c", exec_escaped]
+    else:
+        remote_cmd = []
+
+    cmd = ["ssh", "-p", str(args.port), "localhost"] + remote_cmd
+
+    if args.dry_run:
+        print(" ".join(cmd))
+    else:
+        os.execvp("ssh", cmd)
+
+
+def ssh_server(args, arch, qemuargs, kernelargs):
+    # Check if we need to generate the ssh host keys for the guest.
+    ssh_key_dir = f"{CACHE_DIR}/.ssh"
+    if not os.path.exists(ssh_key_dir):
```

Comment:
About `args.root` you're right, the host keys should be generated at runtime, so we should be able to start sshd in any case.

We won't be able to add the user's keys to `authorized_keys` (unless the ssh keys are also available in the rootfs), and maybe that's fine, as the user can still be able to login using the password. This can probably be detailed later in the README (in the troubleshooting section).

About `--rw` maybe we should simply skip the step that tries to update `authorized_keys`, this should be safe and, again, users can either configure the `authorized_keys` manually or they can still be able to ssh using the password.

I'll do some tests with `--rw` and `--root`. Thanks!